parsing/dialects/dialect1/parserImpls.ftl

```diff
@@ -2259,3 +2261,51 @@ SqlDrop SqlDropMacro(Span s, boolean replace) :
         return SqlDdlNodes.dropMacro(s.end(this), false, id);
     }
 }
+
+List<SqlTableAttribute> CreateJoinIndexTableAttributes() :
+{
+    final List<SqlTableAttribute> list = new ArrayList<SqlTableAttribute>();
+    SqlTableAttribute e;
+}
+{
+    (
+        <COMMA>
+        (
+            e = TableAttributeMap()
+        |
+            e = TableAttributeFallback()
+        |
+            e = TableAttributeChecksum()
+        |
+            e = TableAttributeBlockCompression()
+        ) { list.add(e); }
+    )+
```

Comment:
nit: If you changed the + above to *, I think this could enable you to simplify the logic in other places. For example, you could remove the square brackets around the call to CreateJoinIndexTableAttributes (line 2298), you could declare the tableAttributes variable (line 2290) as final and avoid initalizing it to null (if there are no attributes, it just gets set to an empty list), and then you can remove the null check for tableAttributes from the SqlCreateJoinIndex unparsing function and just do Objects.requireNonNull in the constructor.

Reply:
Yeah I like that, it could help remove some unnecessary checks. I'll work on updating this